src/views/DatasetManager/components/DatasetOverview/components/CategoryAccordion/components/CategoryDetailsDialog/CategoryDetailsDialog.js

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import React, { useCallback, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { Grid, Typography, Dialog, DialogTitle, Box, Card, CardContent, Button, DialogContent } from '@mui/material';
+import { TABLE_DATA_STATUS, ErrorBanner } from '@cosmotech/ui';
+import { GenericTable } from '../../../../../../../../components/';
+import { TranslationUtils } from '../../../../../../../../utils';
+import { KPIValue } from '../../../KPIValue';
+import { useCategoryDetailsDialogHook } from './CategoryDetailsDialogHook';
+
+export const CategoryDetailsDialog = (props) => {
+  const { datasetName, datasetId, getQuery, isDarkTheme, applicationError, clearApplicationErrorMessage } =
+    useCategoryDetailsDialogHook();
+  const { t } = useTranslation();
+
+  const { kpis, category } = props;
+  const [open, setOpen] = useState(false);
+
+  const handleOpen = useCallback(() => {
+    setOpen(true);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    clearApplicationErrorMessage();
+    setOpen(false);
+  }, [clearApplicationErrorMessage]);
+
+  const errorBanner = useMemo(() => {
+    return (
+      <ErrorBanner
+        error={applicationError}
+        labels={{
+          dismissButtonText: t('commoncomponents.banner.button.dismiss', 'Dismiss'),
+          tooLongErrorMessage: t(
+            'commoncomponents.banner.tooLongErrorMessage',
+            'Detailed error message is too long to be displayed. ' +
+              'To read it, please use the COPY button and paste it in your favorite text editor.'
+          ),
+          secondButtonText: t('commoncomponents.banner.button.copy.label', 'Copy'),
+          toggledButtonText: t('commoncomponents.banner.button.copy.copied', 'Copied'),
+        }}
+        clearErrors={clearApplicationErrorMessage}
+      />
+    );
+  }, [applicationError, clearApplicationErrorMessage, t]);
+
+  const categories = useMemo(() => {
```

Comment:
The const should probably be renamed from `categories` to `categoryKpis` or something similar :slightly_smiling_face: